similar/editor/medrobot.cpp

```diff
@@ -87,7 +87,7 @@ namespace dsx {
 static window_event_result robot_dialog_handler(UI_DIALOG *dlg,const d_event &event, robot_dialog *r);
 
 }
-static void call_init_ai_object(object &objp, ai_behavior behavior)
+static void call_init_ai_object(objptridx_t objp, ai_behavior behavior)
```

Comment:
All callers pass a `vobjptridx_t`.  While it can be transparently converted to `objptridx_t`, retaining its original type would document that it must be a valid object.